### PR TITLE
Add new STT mute strategy, accept a set of strategies

### PR DESCRIPTION
**Pull request by markbackman**
#### Please describe the changes in your PR. If it is addressing an issue, please reference that as well.

Prompted by a request from Discord. See Changelog for details.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +41,8 @@ async def on_audio_data(processor, audio, sample_rate, num_channels):
 
 ### Changed
 
+- `STTMuteFilter` now supports multiple simultaneous muting strategies
```
Comment (aconchillo) [inline CHANGELOG.md:44]: tinynit: missing period.

File: src/pipecat/processors/filters/stt_mute_filter.py [hunk 2]
```diff
@@ -67,22 +97,39 @@ async def _handle_mute_state(self, should_mute: bool):
...
+        # Handle function call state changes
+        if isinstance(frame, FunctionCallInProgressFrame):
+            self._function_call_in_progress = True
+            await self._handle_mute_state(await self._should_mute())
+        elif isinstance(frame, FunctionCallResultFrame):
+            self._function_call_in_progress = False
+            await self._handle_mute_state(await self._should_mute())
+
         # Handle bot speaking state changes
         if isinstance(frame, BotStartedSpeakingFrame):
```
Comment (aconchillo) [inline src/pipecat/processors/filters/stt_mute_filter.py:134]: Nit: This could be `elif` now.

**Review by aconchillo — APPROVED** [review]

File: src/pipecat/processors/aggregators/openai_llm_context.py [hunk 3]
```diff
@@ -213,7 +214,8 @@ async def function_call_result_callback(result):
                     arguments=arguments,
                     result=result,
                     run_llm=run_llm,
-                )
+                ),
+                FrameDirection.UPSTREAM,
```
Comment (aconchillo) [inline src/pipecat/processors/aggregators/openai_llm_context.py:218]: I think we should push both UPSTREAM and DOWNTREAM

Comment (aconchillo) [reply src/pipecat/processors/aggregators/openai_llm_context.py:218]: So we basically need to copy all this for both cases.

Comment (aconchillo) [reply src/pipecat/processors/aggregators/openai_llm_context.py:218]: Don't reuse the frame.

File: src/pipecat/processors/aggregators/openai_llm_context.py [hunk 4]
```diff
@@ -196,25 +196,30 @@ async def call_function(
...
+        # Also push a SystemFrame upstream for use by other processors, like STTMuteFilter.
+        progress_frame = FunctionCallInProgressFrame(
+            function_name=function_name,
+            tool_call_id=tool_call_id,
+            arguments=arguments,
+        )
+
+        # Push frame both downstream and upstream
+        await llm.push_frame(progress_frame, FrameDirection.DOWNSTREAM)
+        await llm.push_frame(progress_frame, FrameDirection.UPSTREAM)
```
Comment (aconchillo) [inline src/pipecat/processors/aggregators/openai_llm_context.py:208]: Oh, I put another comment about this before. We should not reuse the same frame.

Comment (markbackman) [reply src/pipecat/processors/aggregators/openai_llm_context.py:208]: Is that for debugging purposes?

Comment (markbackman) [reply src/pipecat/processors/aggregators/openai_llm_context.py]: I saw your comment in Slack. You want unique instances pushed in the frames. 👍

**Review by aconchillo — APPROVED** [review]

**Pull request merged**